File: fence9.cpp
/*
ID: totheso1
LANG: C++14
TASK: fence9
*/

#include <algorithm>
#include <cmath>
#include <cstdlib>
#include <fstream>
#include <iterator>

using namespace std;

static ifstream fin("fence9.in");
static ofstream fout("fence9.out");

#define endl '\n'

template <typename T>
T fin_get() {
  return *istream_iterator<T>(fin);
}

template <typename C>
C fin_get_collection(const int size) {
  C ret;
  copy_n(istream_iterator<typename C::value_type>(fin), size,
         back_inserter(ret));
  return ret;
}

static const int X1 = fin_get<int>();
static const int Y1 = fin_get<int>();
static const int X2 = fin_get<int>();

static int gcd(int a, int b) {
  return a == 0 ? abs(b) : b == 0 ? abs(a) : gcd(b, a % b);
}

static void solve_O_N() {
  int ans = 0;
  for (int y = 1; y < Y1; ++y) {
    ans += static_cast<int>(ceil((X1 - X2) * y / static_cast<double>(Y1))) +
           X2 - 1 - static_cast<int>(floor(X1 * y / static_cast<double>(Y1)));
  }
  fout << ans << endl;
}

static void solve_O_logN() {
  const int boundary_lattice_count = gcd(X1, Y1) + gcd(abs(X2 - X1), Y1) + X2;
  fout << (X2 * Y1 - boundary_lattice_count + 2) / 2 << endl;
}

int main() {
  solve_O_logN();
}

File: fence9.py
"""
ID: totheso1
LANG: PYTHON3
TASK: fence9
"""

from math import ceil, floor, gcd
from typing import Any

fin = open("fence9.in", "r")
fout = open("fence9.out", "w")


def fprint(*args: Any, **kwargs: Any) -> None:
    print(*args, file=fout, **kwargs)


# Code start
X1, Y1, X2 = map(int, fin.readline().split())


def solve_O_N() -> None:
    fprint(
        sum(
            (ceil((X1 - X2) * y / Y1) + X2 - 1) - floor(X1 * y / Y1)
            for y in range(1, Y1)
        )
    )


# Pick's theorem: 2 * Area == 2 * Inside_Lattice_Count + Boundary_Lattice_Count - 2
# Lattice_Count_On_Line_Segment = 1 + gcd(abs(x1 - x2), abs(y1 - y2))
def solve_O_logN() -> None:
    boundary_lattice_count = gcd(X1, Y1) + gcd(abs(X2 - X1), Y1) + X2
    fprint((X2 * Y1 - boundary_lattice_count + 2) // 2)


solve_O_logN()
# Code end
